curvefs/src/client/s3/disk_cache_write.cpp

```diff
@@ -102,10 +104,24 @@ int DiskCacheWrite::UploadFile(const std::string name) {
         posixWrapper_->close(fd);
         return -1;
     }
+    posixWrapper_->close(fd);
+    *buf = buffer;
+    return 0;
+}
+
+int DiskCacheWrite::UploadFile(const std::string name) {
+    uint64_t fileSize;
+    char* buffer = nullptr;
```

Comment:
just allocate buffer in here, because it's deleted in the end of current function.